ListBuilder/DynamicListBuilder.php

```diff
@@ -117,25 +117,28 @@ protected function getMediaUrls($value): string
         return $this->toString($value);
     }
 
-    protected function getMediaUrl(string $value): string
+    private function getMediaUrl(string $value): string
     {
-        return str_replace(urlencode('{id}'), $value, $this->getDownloadUrl());
+        return str_replace('{id}', $value, $this->getDownloadUrl());
     }
 
     /**
      * For performance generate route only once.
      */
-    protected function getDownloadUrl(): string
+    private function getDownloadUrl(): string
     {
         if (null === $this->downloadUrl) {
-            $this->downloadUrl = $this->router->generate(
+            // The given id must be a number which we replace
+            $idReplacerNumber = '875421';
```

Comment:
dont get it - why is this a fixed number?

Reply:
As the route generater is expensive process and should be avoided todo multiple times, its only done once and the number is replaced with `{id}`. In 1.6 it was possible to directly use `{id}` but as the route now requires a number it now used a dummy number to be replaced later with the `{id}` replacer.

Reply:
OK 🙈